apps/emqx/test/emqx_persistent_messages_SUITE.erl

```diff
@@ -88,6 +103,96 @@ t_messages_persisted(_Config) ->
 
     ok.
 
+%% TODO: test quic and ws too
+t_session_subscription_iterators(Config) ->
+    [Node1, Node2] = ?config(nodes, Config),
+    Port = get_mqtt_port(Node1, tcp),
+    Topic = <<"t/topic">>,
+    SubTopicFilter = <<"t/+">>,
+    AnotherTopic = <<"u/another-topic">>,
+    ClientId = <<"myclientid">>,
+    ?check_trace(
+        begin
+            [
+                Payload1,
+                Payload2,
+                Payload3,
+                Payload4
+            ] = lists:map(
+                fun(N) -> <<"hello", (integer_to_binary(N))/binary>> end,
+                lists:seq(1, 4)
+            ),
+            ct:pal("starting"),
+            {ok, Client} = emqtt:start_link([
+                {port, Port},
+                {clientid, ClientId},
+                {proto_ver, v5}
+            ]),
+            {ok, _} = emqtt:connect(Client),
+            ct:pal("publishing 1"),
+            Message1 = emqx_message:make(Topic, Payload1),
+            publish(Node1, Message1),
+            receive_messages(1),
```

Comment:
Q. I must admit this testcase is a bit confusing for me. 😅
1. I wonder why we expect to receive a message here while we're not yet subscribed to anything.
2. Why do we publish with RPC call instead of the usual `emqtt:publish/4`?
3. Why do we expect the feedback message will hit the mailbox in 2 different forms (`{publish, ...}` / `{deliver, ...}`)?

Reply:
1) That particular one has slipped through, shouldn't be there.  I'll make them more assertive.
2) No particular reason.
3) This was copied from elsewhere :upside_down_face: .  Basically, it can take messages from either `emqtt:subscribe` and `emqx:subscribe`.

Reply:
> Basically, it can take messages from either `emqtt:subscribe` or `emqx:subscribe`.

Oh. No wonder it's confusing. I'd pick just one of them, both to be stricter and better illustrate the intention.

Ah, forgot another one.
4. Why `ExpectedMessages = [Message2, Message3]` but not `ExpectedMessages = [Message2, Message3, Message4]` down there?

Reply:
`Message4` targets a topic that's not covered by the subscription.

Reply:
:heavy_check_mark: